### PR TITLE
Ajax comment

### PR DESCRIPTION
**Pull request by AndriiSkavreniuk**
add ajax for comment

### CODE REVIEW

File: library/catalog/static/js/bookDetails.js [hunk 1]
```diff
@@ -1,49 +1,187 @@
...
 
 function saveStarRating(url) {
-    let starValue = getWithExpiry(starStorageKey);
-    if (starValue) {
-        url = url + `${starValue}/`;
+  let starValue = getWithExpiry(starStorageKey);
+  if (starValue) {
+    url = url + `${starValue}/`;
+  }
+  let request = new XMLHttpRequest();
```
Comment (Palusha) [inline library/catalog/static/js/bookDetails.js:58]: What is this request for?

File: library/catalog/static/js/bookDetails.js [hunk 2]
```diff
@@ -1,49 +1,187 @@
...
+  request.open("GET", url, true);
+  ``;
```
Comment (Palusha) [inline library/catalog/static/js/bookDetails.js]: Excessive line

File: library/catalog/static/js/bookDetails.js [hunk 3]
```diff
@@ -1,49 +1,187 @@
...
+  request.send(null);
+  1;
```
Comment (Palusha) [inline library/catalog/static/js/bookDetails.js]: Excessive line

Comment (AndriiSkavreniuk) [reply library/catalog/static/js/bookDetails.js]: > Excessive line

deleted

File: library/catalog/static/js/bookDetails.js (hunk 2 above)
Comment (AndriiSkavreniuk) [reply library/catalog/static/js/bookDetails.js]: > Excessive line

deleted

File: library/catalog/urls.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 urlpatterns = [
     path('home/', views.home, name='library-home'),
     path('registration/', views.registration, name='library-registration'),
-    path('registration_unique_validation/', views.unique_registration_check,
+    path('registration_validation/<str:field_value>', views.unique_registration_check,
```
Comment (Palusha) [inline library/catalog/urls.py]: Change back to `registration_unique_validation/`

File: library/catalog/views.py [hunk 5]
```diff
@@ -6,11 +6,14 @@
...
 from django.contrib.auth import authenticate, login, logout
 from django.contrib.auth.decorators import login_required
+from django.core.serializers import serialize
 from django.db.models import Q
 from django.core.mail import send_mail
+from django.forms import model_to_dict
 from django.http import HttpResponse, JsonResponse
 from django.shortcuts import redirect, render
 from django.urls import reverse
+from mongoengine.queryset.visitor import Q
```
Comment (Palusha) [inline library/catalog/views.py]: We don't need this import any more

File: library/catalog/urls.py (hunk 4 above)
Comment (AndriiSkavreniuk) [reply library/catalog/urls.py]: > Change back to `registration_unique_validation/`

done

File: library/catalog/views.py (hunk 5 above)
Comment (AndriiSkavreniuk) [reply library/catalog/views.py]: > We don't need this import any more

We don't need 'Q' or we don't need 'Q' from this library?

Comment (Palusha) [reply library/catalog/views.py]: We don't need `Q` from this library, there is another slightly higher

**Review by Palusha — APPROVED** [review]

File: library/catalog/static/js/bookDetails.js [hunk 6]
```diff
@@ -1,49 +1,187 @@
...
+    }
+  });
+}
+
+function delete_comment() {
+    let endpointDelete = document.getElementById("commentButton")
+    .getAttribute("url");
+    let csrfmiddlewaretoken = document.getElementsByName("csrfmiddlewaretoken")[0]
+    .value;
+    console.log(endpointDelete)
```
Comment (ivan-dubei) [inline library/catalog/static/js/bookDetails.js]: Remove console log

File: library/catalog/static/js/bookDetails.js [hunk 7]
```diff
@@ -1,49 +1,187 @@
...
+}
+
+function fillCommentsList(list) {
+  let userId = document.getElementById("userId").getAttribute("value");
+  let userRole = document.getElementById("userRole").getAttribute("value");
+
+  const data = list.reviews;
+  $("#commentList").empty();
+
+  for (let i = 0; i < data.length; i++) {
```
Comment (ivan-dubei) [inline library/catalog/static/js/bookDetails.js:82]: Is it the best way to add buttons and comments ?
Is it possible to do this on template level ?

File: library/catalog/templates/book-details.html [hunk 8]
```diff
@@ -5,199 +5,173 @@
...
+{% for message in messages %}
+<div class="toast ">
+    <div class="toast-body font-weight-bold text-danger">
+        <h6>{{ message }}</h6>
+    </div>
+</div>
+
+{% endfor %}
+{% endif %}
+<div style="margin-top: 50px;background-color: #e1e5e7;"
```
Comment (ivan-dubei) [inline library/catalog/templates/book-details.html]: Inline  CSS

File: library/catalog/templates/book-details.html [hunk 9]
```diff
@@ -5,199 +5,173 @@
...
-                                        <svg class="rating__star">
-                                            <use xlink:href="#star"></use>
-                                        </svg>
-                                        <span class="screen-reader">5</span>
-                                    </label>
-                                </div>
-                                <span class="page__hint">Rating</span>
-                            </div>
+    </div>
+    <h2 style="margin-left: 400px">Reviews</h2>
```
Comment (ivan-dubei) [inline library/catalog/templates/book-details.html]: Inline CSS

File: library/catalog/static/js/bookDetails.js (hunk 7 above)
Comment (AndriiSkavreniuk) [reply library/catalog/static/js/bookDetails.js:82]: no
if we use ajax we must add a button only in this way

> Is it the best way to add buttons and comments ?
> Is it possible to do this on template level ?

no
if we use ajax we must add a button only in this way

**Review by ivan-dubei — APPROVED** [review]

**Pull request merged**